plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -55,85 +59,86 @@ public void configure(Binder binder)
         configBinder(binder).bindConfig(OracleConfig.class);
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class)).setBinding().toInstance(ORACLE_MAX_LIST_EXPRESSIONS);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
-        binder.install(new OracleConnectionFactoryModule());
+        newOptionalBinder(binder, RetryStrategy.class).setBinding().to(OracleRetryStrategy.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, Key.get(ConnectionFactory.class, ForRetryingConnectionFactory.class)).setBinding().toProvider(MyProvider.class).in(Scopes.SINGLETON);
     }
 
-    private static class OracleConnectionFactoryModule
-            extends PrivateModule
```

Comment:
Following https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#git-merge-strategy, we should avoid changes in single PR that are reverted (undone) by commit in the same PR.